plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectionCreation.java

```diff
@@ -0,0 +1,163 @@
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static io.trino.plugin.jdbc.H2QueryRunner.createH2QueryRunner;
+import static io.trino.plugin.jdbc.TestingH2JdbcModule.createProperties;
+import static io.trino.tpch.TpchTable.NATION;
+import static io.trino.tpch.TpchTable.REGION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@Test(singleThreaded = true)
+public class TestJdbcConnectionCreation
+        extends AbstractTestQueryFramework
+{
+    private final Map<Connection, Exception> connectionCreations = new ConcurrentHashMap<>();
+    private final AtomicInteger openConnectionCount = new AtomicInteger();
+    private final AtomicInteger closedConnectionCount = new AtomicInteger();
+
+    @BeforeMethod
+    public void verifyAllConnectionsAreClosed()
+    {
+        Optional<Exception> leakedConnection = connectionCreations.values()
+                .stream()
+                .findFirst();
+        if (leakedConnection.isPresent()) {
+            fail("Connection leaked", leakedConnection.get());
+        }
+        assertThat(openConnectionCount.get()).isEqualTo(closedConnectionCount.get());
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createH2QueryRunner(ImmutableList.of(NATION, REGION), createProperties(), new ReusingConnectionH2Module());
+    }
+
+    @Test
+    public void testOpenConnectionsForShowQueries()
+    {
+        assertOpenConnections("SHOW SCHEMAS", 1);
+        assertOpenConnections("SHOW TABLES", 2);
+    }
+
+    @Test
+    public void testOpenConnectionsForTableQueries()
+    {
+        assertOpenConnections("SELECT * FROM nation", 3);
+        assertOpenConnections("CREATE TABLE copy_of_nation AS SELECT * FROM nation", 10);
+        assertOpenConnections("INSERT INTO copy_of_nation SELECT * FROM nation", 11);
+        assertOpenConnections("DROP TABLE copy_of_nation", 3);
+    }
+
+    @Test
+    public void testOpenConnectionsForInformationSchemas()
+    {
+        assertOpenConnections("SELECT * FROM information_schema.schemata", 1);
+        assertOpenConnections("SELECT * FROM information_schema.tables", 1);
+        assertOpenConnections("SELECT * FROM information_schema.columns", 5);
+    }
+
+    @Test
+    public void testOpenConnectionsForPushdownQueries()
+    {
+        assertOpenConnections("SELECT * FROM nation LIMIT 1", 3);
+        assertOpenConnections("SELECT * FROM nation ORDER BY nationkey LIMIT 1", 3);
+        assertOpenConnections("SELECT * FROM nation WHERE nationkey = 1", 3);
+        assertOpenConnections("SELECT avg(nationkey) FROM nation", 3);
+        assertOpenConnections("SELECT * FROM nation n JOIN region r USING(regionkey)", 6);
```

Comment:
`USING` may prevent JOIN pushdown, which looks like something you want to test for https://github.com/trinodb/trino/issues/6781

BTW it would be good to assert the pushdown actually happened.
It won't happen under default configuration due to lack of stats, right?